### ISSUE TITLE
Extracting layer ID from x,y and z coordinates

### ISSUE THREAD

**Issue by Ghallabi** [tomas789/kitti2bag#27]
Hi,

I am working with Kitti dataset and I want to add a "ring id" label to each point . Instead of having XYZI points, I am wondering if we can have XYZRI points ?

**Comment by versatran01** [comment]
When you say "ring id", does that mean which beam each point belongs to? For example, if one is using VLP64, then there would be 64 different ring ids?

**Comment by Ghallabi** [comment]
Yes, exactly. 
Is this label already available ?
Best,

**Comment by versatran01** [comment]
I don't know, but I was trying to do the same thing but looks like it is not possible just from the data.

**Comment by versatran01** [comment]
One could possibly calculate the altitude of each point and then assign a corresponding ring id?

**Comment by aidanjones3** [comment]
I also am interested in determining a solution to this problem. I had two thoughts on the matter: The first being if the kitti datasets somehow have the points ordered already? This may provide some structure to take advantage of. The other would be to look at the timestamp of the points in spherical coordinates per frame. Not entirely sure if this would work but just my thoughts.

**Comment by geoeo** [comment]
Formulae is given in https://github.com/tomas789/kitti2bag/issues/49 But it doesnt seem to work. I would love to hear if it works for you

**Comment by aidanjones3** [comment]
Will attempt, and update you with results.

**Comment by versatran01** [comment]
One (fixable) problem is you won't get clean data. 
I think kitti paper says that they undistorted the point cloud so when you convert them from euclidean to polar coordinates, it will be noisy.

**Comment by geoeo** [comment]
Where do they say that? In the dataset paper? Also, does that mean the synced data part should work?

**Comment by versatran01** [comment]
Yeah I think in the paper. So raw data should work, not synced because that's where they corrected the lidar data.

**Comment by geoeo** [comment]
Thanks for the help. I checked with both and both provide angles which are outside the range of 2-24.8. I looked at the manual again for the HDL 64 E and it states it comes with a calibration.xml. :( Im not really sure how researchers used the kitti velodyne data to provide accurate reconstruction if this is missing

**Comment by versatran01** [comment]
Then it's a problem with kitti.

**Comment by matigolias** [comment]
Check VincentCheungM/Run_based_segmentation#3 for finding the ring in a non-angle based approach

**Comment by mr-abramenko** [comment]
Check https://github.com/Taeyoung96/kitti2bag/commit/a77c6dffe28fc110dca8f7be6167b5b43cfedbff for finding the ring in a angle based approach
